src/wait.ts

```diff
@@ -13,15 +14,15 @@ export async function waitForConfirmation(
   client: Algodv2,
   txid: string,
   waitRounds: number
-): Promise<Record<string, any>> {
+): Promise<PendingTransactionResponse> {
   // Wait until the transaction is confirmed or rejected, or until 'waitRounds'
   // number of rounds have passed.
 
   const status = await client.status().do();
   if (typeof status === 'undefined') {
     throw new Error('Unable to get node status');
   }
-  const startRound = status['last-round'] + 1;
+  const startRound = Number(status.lastRound) + 1;
```

Comment:
We lose precision by converting to a `Number` as rounds are uint64s and the max `Number` is ~2^53. This might cause overflow on things like dev networks, so it seems prudent to use `BigInt` where we can in source code. 

It also seems problematic that our existing APIs take `number` instead of `BigInt` for round numbers... https://github.com/algorand/js-algorand-sdk/blob/bc272cb8048e9cf1ae3b311e2ac17da2ae35ff0f/src/client/v2/algod/block.ts#L11
I think we should change these APIs to accept `BigInt` as well?